### PR TITLE
refactor(typescript): touchable, textbox, tag components

### PR DESCRIPTION
**Pull request by philibea**
## Type

- Refactor

### Summarise concisely:

1. Refactor Touchable, TextBox, Tag to typescript

### CODE REVIEW

File: src/components/Touchable/index.tsx [hunk 1]
```diff
@@ -22,12 +22,23 @@ const styles = {
   `,
 }
 
-const BaseTouchable = ({
-  innerRef,
-  disabled,
+type TouchableProps = {
+  innerRef: React.Ref<Element>
```
Comment (chambo-e) [inline src/components/Touchable/index.tsx:26]:
```suggestion
  innerRef: Ref<Element>
```

File: src/components/Touchable/index.tsx [hunk 2]
```diff
@@ -22,12 +22,23 @@ const styles = {
...
+  children?: React.ReactNode
```
Comment (chambo-e) [inline src/components/Touchable/index.tsx:27]:
```suggestion
  children?: ReactNode
```

File: types/global.d.ts [hunk 3]
```diff
@@ -29,12 +30,12 @@ declare global {
...
     px?: string | number
     py?: string | number
-    verticalAlign?: string
-    width?: string | number
-
     // HTMLAnchorElement.rel
     rel?: string
     // HTMLAnchorElement.target
     target?: string
+    type?: string
```
Comment (chambo-e) [inline types/global.d.ts:37]: Not an xstyled prop :)

File: types/global.d.ts [hunk 4]
```diff
@@ -4,15 +4,16 @@ export {}
...
   type XStyledProps = {
-    backgroundColor?: string
-    as?: string | React.ElementType<unknown>
     align?: string
     alignItems?: string
-    justifyContent?: string
+    as?: string | React.ElementType<unknown>
+    backgroundColor?: string
     color?: string
+    disabled?: boolean
```
Comment (chambo-e) [inline types/global.d.ts:12]: not an xstyled prop

File: src/components/Box/index.tsx [hunk 5]
```diff
@@ -27,8 +28,8 @@ export type BoxProps = {
   children?: ReactNode
   height?: number | string
   width?: number | string
-} & XStyledProps &
-  HTMLAttributes<HTMLElement> & {
+} & (HTMLAttributes<HTMLElement> | Omit<AllHTMLAttributes<HTMLElement>, 'as'>) &
```
Comment (DorianMaliszewski) [inline src/components/Box/index.tsx:31]: I don't like the Omit workaround, as should be typed as React.Element which is union of FunctionComponent, Component and JSX.Element no ?

File: src/components/Tag/index.tsx [hunk 6]
```diff
@@ -108,13 +126,11 @@ Tag.propTypes = {
   isLoading: PropTypes.bool,
   onClose: PropTypes.func,
   textStyle: PropTypes.shape({}),
-  variant: PropTypes.oneOf(Object.keys(variantsContainer)),
+  variant: PropTypes.oneOf(Object.keys(variantsContainer) as [TagVariant]),
```
Comment (DorianMaliszewski) [inline src/components/Tag/index.tsx:129]: TagVariant[] better no ?

Comment (philibea) [reply src/components/Tag/index.tsx]: Yes

**Review by DorianMaliszewski — APPROVED** [review]

File: src/components/Box/index.tsx [hunk 7]
```diff
@@ -27,8 +28,9 @@ export type BoxProps = {
   children?: ReactNode
   height?: number | string
   width?: number | string
-} & XStyledProps &
-  HTMLAttributes<HTMLElement> & {
+} & HTMLAttributes<HTMLElement> &
+  Omit<AllHTMLAttributes<HTMLElement>, 'as'> &
```
Comment (vincentaudebert) [inline src/components/Box/index.tsx:32]: AllHTMLAttributes inclus tous les HTMLAttributes non? On peut virer la ligne du dessus je pense?

**Review by chambo-e — APPROVED** [review]

**Pull request merged**